scripts/pipeline/orthology_benchmark.py

```diff
@@ -17,18 +17,145 @@
+
+
+def find_latest_core(core_names: List[str]) -> str:
+    """Returns the latest core database among the available ones.
+
+    Args:
+        core_names: A list of cores for a species of interest.
+
+    Raises:
+        ValueError: If `core_names` is empty.
+
+    Returns:
+        Name of the latest core database (the latest Ensembl release, the latest version).
+
+    """
+    if len(core_names) == 0:
+        raise ValueError("Empty list of core databases. Cannot determine the latest one.")
+
+    rel_ver = [name.split("_core_")[1].split("_") for name in core_names]
+    rel_ver_int = [list(map(int, i)) for i in rel_ver]
+
+    # There might be two types of versioning
+    # (e.g. caenorhabditis_elegans_core_106_279, caenorhabditis_elegans_core_53_106_279)
+    # Pad with zero(s) on the left (to get e.g. [[0, 106, 279], [53, 106, 279]])
+    max_length = max(len(r) for r in rel_ver_int)
+    pad_token = 0
+    rel_ver_int_pad = [[pad_token] * (max_length - len(r)) + r for r in rel_ver_int]
+
+    rel_ver_arr = numpy.array(rel_ver_int_pad)
+
+    n_cols = rel_ver_arr.shape[1]
+    # Skip columns with any padded values
+    i = max(max_length - len(r) for r in rel_ver_int)
+    while i < n_cols:
+        # Find max value in the i-th column
+        # For the next iteration (i+1) consider only rows where i-th column == max value
+        max_col = numpy.amax(rel_ver_arr[:, i])
+        rows_ind = numpy.where(rel_ver_arr[:, i] == max_col)[0]
+        rel_ver_arr = rel_ver_arr[[rows_ind], :][0]
+        i += 1
+
+    # Trim padded zeros, if any, to get only numbers that appear in the core db name
+    rel_ver_trimmed = numpy.trim_zeros(rel_ver_arr[0], "f")
+    latest_rel_ver = '_'.join(map(str, rel_ver_trimmed))
+    core_name = [core for core in core_names if latest_rel_ver in core][0]
+
+    return core_name
+
+
+def get_core_names(species_names: List[str], host: str, port:int, user: str) -> Dict[str, str]:
+    """Returns the latest core database names for a list of species.
+
+    Args:
+        species_names: Species (genome) names.
+        host: Host for core databases.
+        port: Host port.
+        user: Server username.
+
+    Returns:
+        Dictionary mapping species (genome) names to the latest version of available core names.
+
+    Raises:
+        ValueError: If `species_list` is empty.
+        sqlalchemy.exc.OperationalError: If `user` cannot read from `host:port`.
+
+    """
+    if len(species_names) == 0:
+        raise ValueError("Empty list of species names. Cannot search for core databases.")
+
+    core_names = {}
+
+    eng = create_engine(f"mysql://{user}@{host}:{port}/")
+    result = eng.execute("SHOW DATABASES LIKE '%%_core_%%'").fetchall()
+    all_cores = [i[0] for i in result]
+
+    user_env = os.environ['USER']
+    for species in species_names:
+        core_name = [core for core in all_cores if re.match(f"^({user_env}_)?{species}_core_", core)]
```

Comment:
That's the only reason I can think of. @JAlvarezJarreta wanted to have that in so he might have more scenarios in mind. I can reassure you that `ivana_pilizoticus_core_107_1` won't be ready for the handover 😞

Reply:
I'm sorry to hear it. 🙂 In any case, I suppose the need for testing is sufficient reason for the updated regex.

Reply:
Not only testing. We may have a situation where we copy cores for our usage in `cp?` and those should always have our user prefixed, so I envisioned it may be useful to pick those as well.